platooning_control_ihp/src/platoon_control_ihp.cpp

```diff
@@ -108,51 +92,23 @@ namespace platoon_control_ihp
         ros::CARMANodeHandle::spin();
     }
 
-    bool PlatoonControlIHPPlugin::controlTimerCb()
-    {
-        ROS_DEBUG_STREAM("In control timer callback ");
-        // If it has been a long time since input data has arrived then reset the input counter and return
-        // Note: this quiets the controller after its input stream stops, which is necessary to allow 
-        // the replacement controller to publish on the same output topic after this one is done.
-        long current_time = ros::Time::now().toNSec() / 1000000;
-        ROS_DEBUG_STREAM("current_time = " << current_time << ", prev_input_time_ = " << prev_input_time_ << ", input counter = " << consecutive_input_counter_);
-        if (current_time - prev_input_time_ > config_.shutdownTimeout)
-        {
-            ROS_DEBUG_STREAM("returning due to timeout.");
-            consecutive_input_counter_ = 0;
-            return false;
-        }
 
-        // If there have not been enough consecutive timely inputs then return (waiting for
-        // previous control plugin to time out and stop publishing, since it uses same output topic)
-        if (consecutive_input_counter_ <= config_.ignoreInitialInputs)
-        {
-            ROS_DEBUG_STREAM("returning due to first data input");
-            return false;
+    void  PlatoonControlIHPPlugin::trajectoryPlan_cb(const cav_msgs::TrajectoryPlan::ConstPtr& tp){
+        
+        if (tp->trajectory_points.size() < 2) {
+            ROS_WARN_STREAM("PlatoonControlPlugin cannot execute trajectory as only 1 point was provided");
+            return;
         }
+        cav_msgs::TrajectoryPlanPoint first_trajectory_point = tp->trajectory_points[1]; // TODO this variable appears to be misnamed. It is the second trajectory point
```

Comment:
I am getting more concerned that again we are not working from the same baseline code.  This and the next few lines don't match what is in the baseline (old version in this PR, several lines below on this page).  Yet it looks like you haven't attempted to modify them, as the TODO is left over from several weeks ago.